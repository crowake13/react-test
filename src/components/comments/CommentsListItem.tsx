import React from 'react';
import * as RENDER_LOG from '../../constants/render-log';
import { Comment } from '../../stores/comments/comment.model';
import withRenderLog from '../shared/withRenderLog';

interface CommentListProps extends Comment {
  hideTopBoarder?: boolean;
}

export const CommentListItem = ({
  email,
  name,
  body,
  hideTopBoarder
}: CommentListProps) => {
  return (
    <div className={'mb-2' + (hideTopBoarder ? '' : ' border-top')}>
      <h6 title={email} className="card-subtitle my-2 text-muted">
        {email}
      </h6>
      <h5 title={name} className="card-title">
        {name}
      </h5>
      <p title={body} className="card-text">
        {body}
      </p>
    </div>
  );
};

export default withRenderLog({ greeting: RENDER_LOG.GREETING })(
  CommentListItem
);
