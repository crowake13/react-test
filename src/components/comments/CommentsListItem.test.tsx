import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { Comment } from '../../stores/comments/comment.model';
import { CommentListItem } from './CommentsListItem';

let container: HTMLElement | null = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  if (!container) {
    return;
  }

  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders correctly', () => {
  const postId = 1;
  const comment: Comment = {
    postId,
    id: 1,
    name: 'title 1.1',
    email: 'email 1.1',
    body: 'body 1.1'
  };

  act(() => {
    render(<CommentListItem {...comment} hideTopBoarder={true} />, container);
  });
  expect(container?.innerHTML).toMatchSnapshot();

  act(() => {
    render(<CommentListItem {...comment} />, container);
  });
  expect(container?.innerHTML).toMatchSnapshot();
});
