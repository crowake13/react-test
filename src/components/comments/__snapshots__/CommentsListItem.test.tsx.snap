// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<div class=\\"mb-2\\"><h6 title=\\"email 1.1\\" class=\\"card-subtitle my-2 text-muted line-clamp-1\\">email 1.1</h6><h5 title=\\"title 1.1\\" class=\\"card-title  line-clamp-1\\">title 1.1</h5><p title=\\"body 1.1\\" class=\\"card-text  line-clamp-1\\">body 1.1</p></div>"`;

exports[`renders correctly 2`] = `"<div class=\\"mb-2 border-top\\"><h6 title=\\"email 1.1\\" class=\\"card-subtitle my-2 text-muted line-clamp-1\\">email 1.1</h6><h5 title=\\"title 1.1\\" class=\\"card-title  line-clamp-1\\">title 1.1</h5><p title=\\"body 1.1\\" class=\\"card-text  line-clamp-1\\">body 1.1</p></div>"`;
